appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionFactoryImpl.java

```diff
@@ -600,33 +600,35 @@ private void detectSqlTraceListeners() {
                             _logger.log(Level.SEVERE, "jdbc.sql_trace_listener_cnfe", sqlTraceListener);
                         }
                     }
-                    Class intf[] = listenerClass.getInterfaces();
-                    for (Class interfaceName : intf) {
-                        if (interfaceName.getName().equals("org.glassfish.api.jdbc.SQLTraceListener")) {
-
-                            try {
-
-                                constructors = listenerClass.getConstructors();
-                                for (Constructor constructor : constructors) {
-                                    parameterTypes = constructor.getParameterTypes();
-                                    //For now only the no argument constructors are allowed.
-                                    //TODO should this be documented?
-                                    if (parameterTypes != null && parameterTypes.length == 0) {
-                                        listener = (SQLTraceListener) constructor.newInstance(initargs);
+                    if (listenerClass != null) {
```

Comment:
`listenerClass` can be null which now will not try to register any listener (which would cause a NPE)